tests/test_main.py

```diff
@@ -289,67 +292,57 @@ def test_initialize_github_error(
         """
         Test 'listen' function where suffix is not in handlers
         """
         # Set up return values
-        mock_fedmsg.tail_messages.return_value = [
-            ("dummy", "dummy", "mock_topic", self.old_style_mock_message)
-        ]
+        mock_load_config.return_value = self.mock_config
 
         # Call the function
-        m.listen(self.mock_config)
+        self.old_style_mock_message.topic = "mock_topic"
+        m.callback(self.old_style_mock_message)
 
         # Assert everything was called correctly
         mock_handle_msg.assert_not_called()
 
     @mock.patch(PATH + "handle_msg")
     @mock.patch(PATH + "issue_handlers")
-    @mock.patch(PATH + "fedmsg")
-    def test_listen_no_issue(self, mock_fedmsg, mock_handlers_issue, mock_handle_msg):
+    @mock.patch(PATH + "load_config")
+    def test_listen_no_issue(
+        self, mock_load_config, mock_handlers_issue, mock_handle_msg
+    ):
         """
         Test 'listen' function where the handler returns none
         """
         # Set up return values
         mock_handlers_issue["github.issue.comment"].return_value = None
-        mock_fedmsg.tail_messages.return_value = [
-            ("dummy", "dummy", "d.d.d.github.issue.drop", self.old_style_mock_message)
-        ]
+        mock_load_config.return_value = self.mock_config
 
         # Call the function
-        m.listen(self.mock_config)
+        self.old_style_mock_message.topic = "d.d.d.github.issue.drop"
+        m.callback(self.old_style_mock_message)
 
         # Assert everything was called correctly
         mock_handle_msg.assert_not_called()
 
     @mock.patch(PATH + "handle_msg")
     @mock.patch(PATH + "issue_handlers")
-    @mock.patch(PATH + "fedmsg")
-    def test_listen(self, mock_fedmsg, mock_handlers_issue, mock_handle_msg):
+    @mock.patch(PATH + "load_config")
+    def test_listen(self, mock_load_config, mock_handlers_issue, mock_handle_msg):
         """
         Test 'listen' function where everything goes smoothly
         """
         # Set up return values
         mock_handlers_issue["github.issue.comment"].return_value = "dummy_issue"
-        mock_fedmsg.tail_messages.return_value = [
-            (
-                "dummy",
-                "dummy",
-                "d.d.d.github.issue.comment",
-                self.old_style_mock_message,
-            ),
-            (
-                "dummy",
-                "dummy",
-                "d.d.d.github.issue.comment",
-                self.new_style_mock_message,
-            ),
-        ]
+        mock_load_config.return_value = self.mock_config
 
         # Call the function
-        m.listen(self.mock_config)
+        self.old_style_mock_message.topic = "d.d.d.github.issue.comment"
+        self.new_style_mock_message.topic = "d.d.d.github.issue.comment"
+        m.callback(self.old_style_mock_message)
+        m.callback(self.new_style_mock_message)
```

Comment:
I would put these in two separate code blocks (or otherwise make it clear that these are setups and executions of two separate test scenarios.